packages/labs/virtualizer/src/layouts/flow.ts

```diff
@@ -360,8 +362,13 @@ export class FlowLayout extends BaseLayout<BaseLayoutConfig> {
 
     // If we are scrolling to a specific index or if we are doing another
     // pass to stabilize a previously started reflow, we will already
-    // have an anchor. If not, establish an anchor now.
-    if (this._anchorIdx === null || this._anchorPos === null) {
+    // have an anchor and anchor is in view. If not, establish an anchor now.
+    if (
+      this._anchorIdx === null ||
+      this._anchorPos === null ||
+      this._anchorPos < lower ||
+      this._anchorPos > upper
+    ) {
```

Comment:
While scrolling, sometime the anchor was far away from the current scroll position. There is a fix after to move scrollbar to anchor position: https://github.com/lit/lit/blob/05691ba4848f1bcbec06df7dcd63ee024589ed47/packages/labs/virtualizer/src/layouts/flow.ts#L403
This can move it a lot and makes scroll moving away from mouse cursor. Adding a check to recompute a new anchor here makes scroll much more smooth.